ISMS-631/VT-1072 Add Security Process (#38)
vshn/kb: docs/modules/ROOT/pages/security_vulnerability_process.adoc

> = Security and Vulnerability Process

Reviewer: ```suggestion
= Security and Vulnerability Handling Process
```

> Critical security vulnerabilities are announced by mailing lists from suppliers, security newsletters, MELANI, and many other sources.

Reviewer: ```suggestion
Critical security vulnerabilities are announced on mailing lists from suppliers, on security newsletters, by MELANI, and by many other sources.
```

This security advisories are analysed and if a vulnerability affects the system directly, the update is planned as soon as possible.

> Urgent upgrades are assessed and planned immediately and less urgent vulnerabilities are patched weekly according to our xref:maintenance_process[maintenance process].

Reviewer: ```suggestion
Urgent upgrades are assessed and planned immediately and less urgent vulnerabilities are patched according to our xref:maintenance_process[maintenance process].
```

== General

> Our xref:maintenance_process[maintenance process] ensures weekly system updates (there are some exceptions on customer's systems which are maintained only monthly according to special agreements).

Reviewer: ```suggestion
Our xref:maintenance_process[maintenance process] ensures regular system updates (there are some exceptions on customer's systems which are maintained according to special agreements).
```

We manage server systems based on Ubuntu and RedHat.

> This two big players usually provide updates rapidly.

Reviewer: ```suggestion
These two big players usually provide updates rapidly.
```

> With this regular maintenance the known vulnerabilities are eliminated withing one week.

Reviewer: ```suggestion
With this regular maintenance the known vulnerabilities are eliminated within one week.
```

Internal security incidents are reported according our ISO27001 certified https://handbook.vshn.ch/hb/domain_ism_incidentmanagement.html[incident management process].

Our https://handbook.vshn.ch/hb/role_responsibleops.html[Responsible Ops] engineers are supervising our monitoring dashboard during office hours according to our xref:monitoring_concept[monitoring concept].
The Responsible Ops engineers are able to react quickly during office hours.

Customers with 24/7 support contracts can call the on-call engineer anytime.

> Customers with SLA premium can define services which would alerting the on-call engineer in case of critical status.

Reviewer: Do we need these lines? Isn't that a duplication what we already document in our SLA document? Duplication leads in to the issue of having outdated document sooner or later.

Author: Ah yes, you're right, in this section it is really a bit duplication.

Do you think we should remove it or change it to something like below?

```suggestion
Customers can report vulnerabilities to VSHN according their https://kb.vshn.ch/products/sla_en.html[Service Level Agreement].
```

Off-topic: Where is is the legal docs repo? Looks like it is not here in the kb repo and I have to link it with the complete URL.

Reviewer: Yes, that makes sense.

> Where is is the legal docs repo?

https://git.vshn.net/vshn/kb.vshn.net/-/tree/master/docs/modules/ROOT/pages

We regularly discuss security issues in our https://vshn.chat[vshn.chat] #security channel.

== Security Process

=== Office Hours

Our standard procedure to handle security issues or vulnerabilities.

. Someone (customer, employee, https://handbook.vshn.ch/hb/role_responsibleops.html[Responsible Ops], etc.) become alert of a vulnerability.
. Ticket is created in https://control.vshn.net.
. If it's a critical vulnerability it's announced in our chat and an instant task force starts discussing potential solution paths.
. If an immediate action within 24h is required, the affected systems will be updated. Because of our flexible working philosophy there is always someone found who can handle the update.
. Such maintenance work is announced via:

> .. https://status.vshn.net[status.vshn.net]
> .. https://status.appuio.ch/[status.appuio.ch]

Reviewer: No need for the `[]` when the URL is the same as the text.

```suggestion
.. https://status.vshn.net/
.. https://status.appuio.ch/
```

=== Outside Office Hours

VSHN doesn't actively supervise vulnerability announcements outside office hours.

Customers with a 24/7 contract who become aware of a critical vulnerability should...

. xref:create_ticket[create a ticket] in https://control.vshn.net/tickets[control.vshn.net] with high priority selected
. AND call our 24/7 on-call number.

The on-call engineer will assess the issue and has the possibility to call 2nd level on-call engineers, if there is a bigger problem to tackle down.

== Lists

We're subscribed to the most important security announcement lists, such as:

* OS
** Ubuntu https://lists.ubuntu.com/mailman/listinfo/ubuntu-security-announce[ubuntu-security-announce] mailing list

> ** Redhat https://www.redhat.com/wapps/ugc/protected/notif.html[Errata Notifications]

Reviewer: ```suggestion
** Red Hat https://www.redhat.com/wapps/ugc/protected/notif.html[Errata Notifications]
```
